software/speller/data_collection_platform/dcp/api.py

```diff
@@ -46,6 +46,83 @@ def char_pressed():
         user_char_pressed = request.form['char_pressed']
 
 
+@bp.route('/collection/start', method=['POST'])
+def collection_start():
+    """
+    Starts the collection of data as the frontend
+    starts highlighting a character
+
+    Takes a JSON attachment with the following shape:
+    {
+        "data": {
+            "character":    <character>,
+            "phase":        <float>,
+            "frequency"     <float>:
+        }
+    }
+    """
+
+    # TODO ensure JSON data structure is compatible
+
+    json_data = request.get_json()
+
+    # TODO write function to validate datatype
+
+    with shared.collecting.get_lock():
+        shared.collecting.value = True
+
+    with shared.character.get_lock():
+        shared.character.value = json_data["data"]["character"]
+
+    with shared.frequency.get_lock():
+        shared.character.value = json_data["data"]["frequency"]
+
+    with shared.phase.get_lock():
+        shared.character.value = json_data["data"]["phase"]
+
+    return 200
```

Comment:
Laurence, please add some checks that json_data contains the three required fields, if not return the correct response code (e.g. 400)